src/__tests__/data/decoratedModels/movies.ts

```diff
@@ -0,0 +1,71 @@
+import { TigrisCollection } from "../../../decorators/tigris-collection";
+import { TigrisPrimaryKey } from "../../../decorators/tigris-primary-key";
+import { TigrisDataTypes, TigrisSchema } from "../../../types";
+import { TigrisField } from "../../../decorators/tigris-field";
+
+/******************************************************************************
+ * `Movie` class demonstrates a Tigris collection schema generated using
+ * decorators. All the collection fields are explicitly typed.
+ *****************************************************************************/
+export class Actor {
+	@TigrisField(TigrisDataTypes.STRING, {maxLength: 64})
+	firstName: string;
+
+	@TigrisField(TigrisDataTypes.STRING, {maxLength: 64})
+	lastName: string;
+}
+
+@TigrisCollection("movies")
+export class Movie{
+
+	@TigrisPrimaryKey(TigrisDataTypes.STRING, {autoGenerate: true, order: 1})
+	movieId: string;
+
+	@TigrisField(TigrisDataTypes.STRING)
+	title: string;
+
+	@TigrisField(TigrisDataTypes.INT32)
+	length: number;
+
+	@TigrisField(TigrisDataTypes.ARRAY, {subType: Actor})
```

Comment:
I don't know decorators but would there be a way of doing this instead:
```
	@TigrisField(TigrisDataTypes.ARRAY<Actor>)
```
That feels a little cleaner.

Reply:
Generics are exactly what we looking for, just that it cannot be inferred using reflection. TS Reflection apis are very limited.